docs/user-guide/02-conventional-features.md

```diff
@@ -792,3 +792,394 @@ In this example, `globalUniform` is appended to the global root signature / pipe
+    //
+    //     c = in_c;
+    // }
+};
+
+struct GenerateCtorInner2
+{
+    internal int a;
+    /// Implicitly generate:
+    // internal __init(int in_a)
+    // {
+    //     a = in_a;
+    // }
+};
+struct GenerateCtor2 : GenerateCtorInner2
+{
+    internal int b;
+    public int c;
+
+    /// Note: `internal b` is missing init expression,
+    // Do not generate a `public` 'member-wise' constructor.
+
+    /// Implicitly generate:
+    // internal __init(int in_a, int in_b, int in_c)
+    // {
+    //     this = GenerateCtorInner2(in_a);
+    //
+    //     b = in_b;
+    //     c = in_c;
+    // }
+};
+```
+
+Initializer Lists
+----------
+Initializer List's are an expression of the form `{...}`.
+
+This form is **different** from declaring a new scope
+```c#
+int myFunc()
+{
+    { // New Scope
+        int a = {}; // Initializer List
+    }
+}
+```
+
+### Initializer List's - Scalar
+
+```c#
+// Equivalent to `int a = 1`
+int a = {1};
+```
+
+### Initializer List's - Vectors
+
+```c#
+// Equivalent to `float3 a = float3(1,2,3)`
+float3 a = {1, 2, 3};
+```
+
+### Initializer List's - Arrays/Matrixes
+
+#### Array Of Scalar's
+
+```c#
+// Equivalent to `int[2] a; a[0] = 1; a[1] = 2;`
+int a[2] = {1, 2}
+```
+
+#### Array Of Aggregate's
+
+```c#
+// Equivlent to `float3 a[2]; a[0] = {1,2,3}; b[1] = {4,5,6};`
+float3 a[2] = {{1,2,3}, {4,5,6}};
+```
+#### C-Style-Initializer-List
+
+```c#
+// Equivalent to `float3 a[2] = {{1,2,3}, {4,5,6}};`
+float3 a[3] = {1,2,3, 4,5,6}; 
+```
+
+### Initializer Lists - Struct
+
+Slang has 2 seperate calling style for an Initializer List with `struct`'s:
+1. Initializer-list may call a constructor
+    * Can also call auto-generated constructors
+```c#
+struct GenerateCtorInner1
+{
+    internal int a = 0;
+    
+    /// Implicitly generate:
+    // internal __init(int in_a)
+    // {
+    //     a = 0;
+    //
+    //     a = in_a;
+    // }
+
+    static GenerateCtorInner1 callGenerateCtorInner1()
+    {
+        // Calls `GenerateCtorInner1::__init(1);`
+        return {1};
+    }
+};
+struct GenerateCtor1 : GenerateCtorInner1
+{
+    internal int b = 0;
+    public int c;
+
+    /// Implicitly generate:
+    // internal __init(int in_a, int in_b, int in_c)
+    // {
+    //     this = GenerateCtorInner1(in_a);
+    //
+    //     b = 0;
+    //
+    //     b = in_b;
+    //     c = in_c;
+    // }
+    //
+    // public __init(int in_c)
+    // {
+    //     this = GenerateCtorInner1();
+    //
+    //     b = 0;
+    //
+    //     c = in_c;
+    // }
+    static GenerateCtorInner1 callInternalGenerateCtor()
+    {
+        // Calls `GenerateCtor1::__init(1, 2, 3);`
+        return {1, 2, 3};
+    }
+    static GenerateCtorInner1 callPublicGenerateCtor()
+    {
+        // Calls `GenerateCtor1::__init(1);`
+        return {1}; 
+    }
+};
+
+...
+
+// Calls `{ GenerateCtor1::__init(3), GenerateCtor1::__init(2) }`
+GenerateCtor1 val[2] = {{ 3 }, { 2 }};
+```
+
+2. C-Style-Initializer-List
```

Comment:
"In addition, slang allows provides compatbility support for legacy C-style structs. When the struct type qualifies as a "C-Style" struct, it is allowed to provide less arguments than struct members in an initilaizer list.

A struct is considered a C-style struct if:
- condition1
- condition2
- condition3

For example,
```
code
```
".